### PR TITLE
To Do list: add & remove

### PR DESCRIPTION
**Pull request by rubydevi**
1. I created a new JavaScript file to implement the new functionality.
2. I removed all hardcoded items from the tasks array.
3. I implemented a function to add a new task. This function added a new element to the array.
4. I implemented a function to delete a task. This function removed an element from the array.
5. I implemented a function to edit task descriptions.
6. By default, when adding a new task, it had the property completed set to false, and the property index set to the value of the new array length.
7. When deleting a task, I updated the indexes of all the remaining tasks to represent the current list order and ensured uniqueness.
8. I ensured that all changes made to the To-Do List were saved in local storage.

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -17,11 +17,14 @@ <h2>Today's To Do</h2>
...
-            <div id="todo-list"></div>
+            <div id="todoList"></div>
             <div class="list-footer">
                 <button>Clear all completed</button>
             </div>
         </div>
     </main>
+
+    <!-- <script src="index.js"></script>
+    <script src="todoList.js"></script> -->
```
Comment (binhussen) [inline src/index.html]: - Kindly, Do not commit old pieces of code as inline comments. They will make your project look messy. If you need to review a previous version of your code, you can always use git history. 👍

File: src/index.js [hunk 2]
```diff
@@ -1,36 +1,36 @@
...
-  });
-};
-
-window.addEventListener('DOMContentLoaded', renderTodoList);
+import {
+  renderTodoList, addTask, editTaskDescription, loadTasksFromLocalStorage, saveTasksToLocalStorage,
+} from './modules/todoList.js';
+
+window.addEventListener('DOMContentLoaded', loadTasksFromLocalStorage);
+window.addEventListener('beforeunload', saveTasksToLocalStorage);
```
Comment (binhussen) [inline src/index.js:7]: - Kindly, you are storing to local storage when `DOMContentLoaded`, so make it update when add, edit, or remove operation is done to improve user experience 👍

File: src/modules/todoList.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+      const deleteButtons = document.querySelectorAll('.delete-button');
+      const dragRows = document.querySelectorAll('.drag-row');
+
+      // Hide all the existing delete buttons and show all the drag buttons
+      deleteButtons.forEach((button) => {
+        button.style.display = 'none';
+      });
+      dragRows.forEach((row) => {
+        row.style.display = 'inline-block';
+      });
```
Comment (binhussen) [inline src/modules/todoList.js:76]: - Kindly, the delete operation is not updating the `index` which leads to having duplicate `index` so when each item has been deleted update the hole todo list index to fit the requirement 👍

**Review by binhussen — CHANGES_REQUESTED** [review]
Hi @rubydevi , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

- Well-structured Code ✔️
- Add, Edit & Delete Works ✔️
- Used GitHub flow ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by julie-ify — APPROVED** [review]
Hi @rubydevi,

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**